### PR TITLE
update doxygen style according to ros2 core standard

### LINKED ISSUE #202: update doxygen style to ros2 core
We're currently using a different style of doxygen than the rest of the ros2 core packages. We should adapt their style in the form of 
```
/// Some brief instruction
/**
 * more details, more description
 * 
 * \param [in/out] <type> description
 * \return <type> description
 */
void some_func();
```

examples can be found here: https://github.com/ros2/rclcpp/blob/master/rclcpp/include/rclcpp/node.hpp#L79-L84

### PR DESCRIPTION
**Pull request by BorgesJVT**
Fix #202.

### CODE REVIEW

File: controller_manager/include/controller_manager/controller_manager.hpp [hunk 1]
```diff
@@ -73,10 +73,11 @@ class ControllerManager : public rclcpp::Node
     const std::string & controller_name,
     const std::string & controller_type);
 
+  /// load_controller loads a controller by name, the type must be defined in the parameter server.
   /**
-   * @brief load_controller loads a controller by name, the type must be defined in the parameter server
-   * @param controller_name as a string.
-   * @see Documentation in controller_manager_msgs/LoadController.srv
+   * \param[in] controller_name as a string.
+   * \return controller_interface::ControllerInterfaceSharedPtr controller
```
Comment (bmagyar) [inline controller_manager/include/controller_manager/controller_manager.hpp:79]: I reiterating the type name here extremely redundant and prone to breaking. Doxygen knows how to generate that part,we should only supply text that explains what the code may not. 
@Karsten1987 is there a less verbose version you'd recommend?

**Review by bmagyar — CHANGES_REQUESTED** [review]
Thanks for this! Let's check styles first w @Karsten1987, if really like to avoid reiterating full C++ typenames in docstrings.

File: controller_manager/include/controller_manager/controller_manager.hpp (hunk 1 above)
Comment (BorgesJVT) [reply controller_manager/include/controller_manager/controller_manager.hpp:79]: So, should I remove the <type> from the style?

Just for notice, It is already exists C++ typenames in docstrings in some files:
https://github.com/ros-controls/ros2_control/blob/master/hardware_interface/include/hardware_interface/actuator_interface.hpp#L95
https://github.com/ros-controls/ros2_control/blob/master/hardware_interface/src/component_parser.cpp#L111

Comment (bmagyar) [reply controller_manager/include/controller_manager/controller_manager.hpp:79]: Yeah I am aware we already have precedent for it but precedent doesn't mean it's right ;)  Let's wait for Karsten's take on this before you spend time

File: hardware_interface/include/hardware_interface/hardware_info.hpp [hunk 2]
```diff
@@ -24,88 +24,88 @@ namespace hardware_interface
 {
 
 /**
- * \brief This structure stores information about components defined for a specific hardware
+ * /// This structure stores information about components defined for a specific hardware
```
Comment (Karsten1987) [inline hardware_interface/include/hardware_interface/hardware_info.hpp:27]:
```suggestion
/// This structure stores information about components defined for a specific hardware
```
this looks odd. One line comments should only be with `///`

Comment (Karsten1987) [reply hardware_interface/include/hardware_interface/hardware_info.hpp:27]: same below.

File: hardware_interface/include/hardware_interface/sensor_interface.hpp [hunk 3]
```diff
@@ -29,7 +29,7 @@ namespace hardware_interface
 {
 
 /**
-  * \brief Virtual Class to implement when integrating a stand-alone sensor into ros2_control.
+  * /// Virtual Class to implement when integrating a stand-alone sensor into ros2_control.
```
Comment (Karsten1987) [inline hardware_interface/include/hardware_interface/sensor_interface.hpp:32]: please extract the brief line as a first line of the doc block

File: joint_limits_interface/include/joint_limits_interface/joint_limits_interface.hpp [hunk 4]
```diff
@@ -108,7 +108,7 @@ class JointLimitHandle
   double prev_pos_;
   double prev_vel_;
 
-  /** \brief Return velocity for limit calculations.
+  /** /// Return velocity for limit calculations.
```
Comment (Karsten1987) [inline joint_limits_interface/include/joint_limits_interface/joint_limits_interface.hpp:111]: might that be a copy-paste error?

File: controller_manager/include/controller_manager/controller_manager.hpp (hunk 1 above)
Comment (Karsten1987) [reply controller_manager/include/controller_manager/controller_manager.hpp:79]: I agree, I don't necessarily think we have to mention the data type here.

**Review by Karsten1987 — CHANGES_REQUESTED** [review]
Thanks a bunch for picking this up. Really valuable. 
I think there are few occasions thought where maybe the copy-paste didn't really hit the nail. Please re-address them.

File: controller_manager/include/controller_manager/controller_manager.hpp (hunk 1 above)
Comment (BorgesJVT) [reply controller_manager/include/controller_manager/controller_manager.hpp]: Ok, I removed the type from the style.

File: joint_limits_interface/include/joint_limits_interface/joint_limits_interface.hpp (hunk 4 above)
Comment (BorgesJVT) [reply joint_limits_interface/include/joint_limits_interface/joint_limits_interface.hpp]: Yes.

File: hardware_interface/include/hardware_interface/sensor_interface.hpp (hunk 3 above)
Comment (BorgesJVT) [reply hardware_interface/include/hardware_interface/sensor_interface.hpp]: I changed to `///` all single line doc sentences.

**Review by bmagyar — APPROVED** [review]

**Comment by bmagyar** [comment]
@Karsten1987 please give it another go, I'm happy for it to go ahead

File: joint_limits_interface/include/joint_limits_interface/joint_limits_interface.hpp [hunk 5]
```diff
@@ -38,7 +38,7 @@
 namespace joint_limits_interface
 {
 
-/** \brief The base class of limit handles for enforcing position, velocity, and effort limits of
+/** The base class of limit handles for enforcing position, velocity, and effort limits of
```
Comment (Karsten1987) [inline joint_limits_interface/include/joint_limits_interface/joint_limits_interface.hpp:41]:
```suggestion
/**
 * The base class of limit handles for enforcing position, velocity, and effort limits of
```

File: joint_limits_interface/include/joint_limits_interface/joint_limits_interface.hpp [hunk 6]
```diff
@@ -330,7 +325,7 @@ class PositionJointSoftLimitsHandle : public JointSoftLimitsHandle
   }
 };
 
-/** \brief A handle used to enforce position, velocity, and effort limits of an effort-controlled
+/** A handle used to enforce position, velocity, and effort limits of an effort-controlled
```
Comment (Karsten1987) [inline joint_limits_interface/include/joint_limits_interface/joint_limits_interface.hpp:328]:
```suggestion
/**
 * A handle used to enforce position, velocity, and effort limits of an effort-controlled
```

File: joint_limits_interface/include/joint_limits_interface/joint_limits_interface.hpp [hunk 7]
```diff
@@ -555,7 +549,7 @@ class VelocityJointSaturationHandle : public JointLimitHandle
   }
 };
 
-/** \brief A handle used to enforce position, velocity, and acceleration limits of a
+/** A handle used to enforce position, velocity, and acceleration limits of a
```
Comment (Karsten1987) [inline joint_limits_interface/include/joint_limits_interface/joint_limits_interface.hpp:552]:
```suggestion
/**
 * A handle used to enforce position, velocity, and acceleration limits of a
```

**Review by Karsten1987 — APPROVED** [review]
thanks again for the patch.

**Comment by Karsten1987** [comment]
waiting on linters. then merge.

**Pull request merged**